Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -405,8 +405,8 @@ if(BUILD_SWIG_PYTHON_SUPPORT OR BUILD_NE
 
     string(REGEX MATCH "[0-9].[0-9]" PYTHON_VERSION_MAJOR_MINOR "${PYTHON_VERSION_FULL}")
     if(UNIX)
-        set(PYTHON_PLUGIN_INSTALL_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages/opencv)
-        set(PYTHON_PACKAGES_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages)
+        set(PYTHON_PLUGIN_INSTALL_PATH lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages/opencv)
+        set(PYTHON_PACKAGES_PATH lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages)
     endif()
     if(WIN32)
         get_filename_component(PYTHON_PATH "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\${PYTHON_VERSION_MAJOR_MINOR}\\InstallPath]" ABSOLUTE CACHE)
@@ -731,7 +731,7 @@ configure_file("${CMAKE_CURRENT_SOURCE_D
 # Set CMAKE_INCLUDE_DIRS_CONFIGCMAKE to the list of include directories:
 
 set(CMAKE_INCLUDE_DIRS_CONFIGCMAKE "${CMAKE_INSTALL_PREFIX}/include/opencv")
-set(CMAKE_LIB_DIRS_CONFIGCMAKE "${CMAKE_INSTALL_PREFIX}/lib")
+set(CMAKE_LIB_DIRS_CONFIGCMAKE "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
 
 #exec_program(${CMAKE_COMMAND} ARGS "-E make_directory \"${CMAKE_BINARY_DIR}/unix-install/\"")
 exec_program(mkdir ARGS "-p \"${CMAKE_BINARY_DIR}/unix-install/\"")
@@ -764,19 +764,19 @@ endif()
 #  Part 2/2: ${BIN_DIR}/unix-install/opencv.pc -> For use *with* "make install"
 # prefix=/usr
 # exec_prefix=${prefix}
-# libdir=${exec_prefix}/lib
+# libdir=${exec_prefix}/lib${LIB_SUFFIX}
 # includedir=${prefix}/include/opencv
 # -------------------------------------------------------------------------------------------
 set(prefix 		${CMAKE_INSTALL_PREFIX})
 set(exec_prefix		"\${prefix}")
-set(libdir		"\${exec_prefix}/lib")
+set(libdir		"\${exec_prefix}/lib${LIB_SUFFIX}")
 set(includedir		"\${prefix}/include")
 set(VERSION             ${OPENCV_VERSION})
 
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/opencv.pc.cmake.in" "${CMAKE_BINARY_DIR}/unix-install/opencv.pc" @ONLY IMMEDIATE)
 
 if(UNIX)
-    install(FILES ${CMAKE_BINARY_DIR}/unix-install/opencv.pc DESTINATION lib/pkgconfig)
+    install(FILES ${CMAKE_BINARY_DIR}/unix-install/opencv.pc DESTINATION lib${LIB_SUFFIX}/pkgconfig)
 endif()
 
 
Index: 3rdparty/flann/CMakeLists.txt
===================================================================
--- 3rdparty/flann/CMakeLists.txt.orig
+++ 3rdparty/flann/CMakeLists.txt
@@ -86,8 +86,8 @@ ENDIF(WIN32)
 INSTALL (
     TARGETS flann flann_s
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
 )
 
 # INSTALL (
Index: interfaces/ffopencv/CMakeLists.txt
===================================================================
--- interfaces/ffopencv/CMakeLists.txt.orig
+++ interfaces/ffopencv/CMakeLists.txt
@@ -33,5 +33,5 @@ set_target_properties(${the_target} PROP
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
     )
Index: src/cv/CMakeLists.txt
===================================================================
--- src/cv/CMakeLists.txt.orig
+++ src/cv/CMakeLists.txt
@@ -66,8 +66,8 @@ add_dependencies(${the_target} cxcore)
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${lib_hdrs}
     DESTINATION include/opencv
Index: src/cvaux/CMakeLists.txt
===================================================================
--- src/cvaux/CMakeLists.txt.orig
+++ src/cvaux/CMakeLists.txt
@@ -76,8 +76,8 @@ add_dependencies(${the_target} cxcore cv
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${aux_hdrs_ext}
         DESTINATION include/opencv
Index: src/cxcore/CMakeLists.txt
===================================================================
--- src/cxcore/CMakeLists.txt.orig
+++ src/cxcore/CMakeLists.txt
@@ -66,8 +66,8 @@ add_dependencies(${the_target} opencv_la
 
 install(TARGETS ${the_target}
     RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 # install each module headers separately
 install(FILES ${lib_hdrs}
Index: src/highgui/CMakeLists.txt
===================================================================
--- src/highgui/CMakeLists.txt.orig
+++ src/highgui/CMakeLists.txt
@@ -191,8 +191,8 @@ endif()
 if(WIN32)
 	target_link_libraries(${the_target} comctl32 gdi32 ole32)
 	
-	if (MSVC)
-		target_link_libraries(${the_target} vfw32)
+	if (MSVC)
+		target_link_libraries(${the_target} vfw32)
 	endif()
 
 	if(MINGW)
@@ -206,8 +206,8 @@ endif()
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${highgui_ext_hdrs}
         DESTINATION include/opencv
Index: src/ml/CMakeLists.txt
===================================================================
--- src/ml/CMakeLists.txt.orig
+++ src/ml/CMakeLists.txt
@@ -61,8 +61,8 @@ add_dependencies(${the_target} cxcore)
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${lib_hdrs}
         DESTINATION include/opencv
Index: tests/cxts/CMakeLists.txt
===================================================================
--- tests/cxts/CMakeLists.txt.orig
+++ tests/cxts/CMakeLists.txt
@@ -63,6 +63,6 @@ target_link_libraries(${the_target} ${OP
 if(WIN32)
 install(TARGETS ${the_target}
     RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 endif()

 	  	 
