diff -crB cmus-2.2.0.orig/mp4.c cmus-2.2.0/mp4.c
*** cmus-2.2.0.orig/mp4.c	2010-01-08 11:06:09.423202273 +0100
--- cmus-2.2.0/mp4.c	2010-01-08 11:15:28.829202191 +0100
***************
*** 22,28 ****
  #include "debug.h"
  #include "file.h"
  
! #include <mp4.h>
  #include <faad.h>
  
  #include <errno.h>
--- 22,28 ----
  #include "debug.h"
  #include "file.h"
  
! #include <mp4v2/mp4v2.h>
  #include <faad.h>
  
  #include <errno.h>
